library/hkdf.c

```diff
@@ -56,11 +61,13 @@ int mbedtls_hkdf_extract( const mbedtls_md_info_t *md,
                           const unsigned char *ikm, size_t ikm_len,
                           unsigned char *prk )
 {
-    unsigned char null_salt[MBEDTLS_MD_MAX_SIZE] = { '\0' };
-
+    const unsigned char null_salt[MBEDTLS_MD_MAX_SIZE] = { '\0' };
+    size_t hash_len;
     if( salt == NULL )
     {
-        size_t hash_len;
```

Comment:
Declaring variables at the beginning of the if scope is fine with the ISO C90, did you receive any error with this code? If so, could you please provide the tools, and flags used, and preferably, the error itself?